.buildkite/data.rayci.yml

```diff
@@ -47,6 +57,27 @@ steps:
         --except-tags data_integration,doctest
     depends_on: data14build
 
+  - label: ":database: data: python {{matrix.python}} tests ({{matrix.worker_id}})"
+    if: build.env("RAYCI_CONTINUOUS_BUILD") == "1" || build.pull_request.labels includes "continuous-build"
+    tags: 
+      - python
+      - data
+    instance_type: medium
+    commands:
+      - bazel run //ci/ray_ci:test_in_docker -- //python/ray/data/... //python/ray/air/... data
+        --workers 2 --worker-id {{matrix.worker_id}} --parallelism-per-worker 3
+        --test-env RAY_DATA_USE_STREAMING_EXECUTOR=1
+        --except-tags data_integration,doctest
+        --python-version {{matrix.python}}
+    depends_on: databuild-multipy
+    matrix:
+      setup:
+        python:
+          - "3.8"
+        worker_id:
+          - "0"
+          - "1"
```

Comment:
nit: make it one line?

I think we should support inherent matrix+sharding support in rayci now..